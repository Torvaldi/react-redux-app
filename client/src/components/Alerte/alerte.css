/* BLOCK */
.alertBlockError, .alertBlockSuccess, .alertBlockInfo {
    width: 100%;
    border-radius: 2px;
    padding: 10px 0px;
    margin: 10px 0px;
    -webkit-box-shadow: 0px 0px 2px 0px rgba(0,0,0,0.75);
    -moz-box-shadow: 0px 0px 2px 0px rgba(0,0,0,0.75);
    box-shadow: 0px 0px 2px 0px rgba(0,0,0,0.75);
}

.alertBlockError {
    background-color: #e57373;
}

.alertBlockSuccess {
    background-color: #81c784;
}

.alertBlockInfo {
    background-color: var(--secondary);
}

/* TEXT */
.alertTextError, .alertTextSuccess, .alertTextInfo {
   margin-left: 10px;
   font-weight: bold;
}

.alertTextError {
    color: #b71c1c;
}

.alertTextSuccess {
    color: #1b5e20;
}

.alertTextInfo {
    color: white;
}

