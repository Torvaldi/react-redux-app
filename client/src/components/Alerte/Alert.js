import React from 'react';
import propTypes from 'prop-types';
import './alerte.css';

const Alert = (props) => {
 
  let blockClass;
  let textClass;
  // zero for error
  if(props.type === 0){
    blockClass = 'alertBlockError';
    textClass = 'alertTextError';
  }

  // 1 for sucess
  if(props.type === 1){
    blockClass = 'alertBlockSuccess';
    textClass = 'alertTextSuccess';
  }

  // 1 for informative message
  if(props.type === 2){
    blockClass = 'alertBlockInfo';
    textClass = 'alertTextInfo';
  }

  return (
    <article className={blockClass}>
      <span className={textClass}>{props.message}</span>
    </article>
  );
}

Alert.propTypes = {
  message: propTypes.string.isRequired,
  type: propTypes.number.isRequired,
}

export default Alert;
