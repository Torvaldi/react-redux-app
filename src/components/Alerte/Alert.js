import React from 'react';

import './alerte.css';

const Alert = (props) => {
  return (
    <article className="alertBlock">
      <span className="alertText">{props.message}</span>
    </article>
  );
}

export default Alert;
