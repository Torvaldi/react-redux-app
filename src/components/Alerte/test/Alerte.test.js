import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Alerte from '../Alert';

configure({ adapter: new Adapter() });

describe('<Alerte />', () => {
    // snapshot
    it('should render Alerte Componenent', () => {
        const message = "hi there";
        const wrapper = shallow(<Alerte message={message} />);
        expect(wrapper).toMatchSnapshot();
    });
    
    // props
    it('should render with props message', () => {
        const message = "yo";
        const AuthLayoutComponent = mount(<Alerte message={message} />);

        expect(AuthLayoutComponent.prop('message')).toEqual(message);
    });
    
});