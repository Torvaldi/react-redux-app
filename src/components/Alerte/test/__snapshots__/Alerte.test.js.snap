// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alerte /> should render Alerte Componenent 1`] = `ShallowWrapper {}`;
